# Recognizing Patterns in Crime

For this analysis we will be looking at crime in Montgomery county Maryland. Using this data we can hopefully answer a few questions.

Using the dates we can determine the following:

1) During what month are most crimes committed?
2) During what time of the day are most crimes committed?
3) What day of the week are most likely to NOT be committed? 




In [43]:
# Let's first read in the data and clean it up a bit.

import pandas as pd
import datetime as dt

crime_data = pd.read_csv("MontgomeryCountyCrime2013.csv")
crime_data['Dispatch Date / Time'] = pd.to_datetime(crime_data['Dispatch Date / Time'])

# Print out the # of crimes committed in each month
month_of_crime = crime_data['Dispatch Date / Time'].dt.month
month_of_crime_sorted = month_of_crime.value_counts().sort_index()
print("The month's of crime committed: ")
print(month_of_crime_sorted)

# Print out the number of crimes that occur in the day and night 
# day is defined as the time between 6am - 6pm.
hour_crime = crime_data['Dispatch Date / Time'].dt.hour
day_crime = (hour_crime >= 6) & (hour_crime < 18)

# print out the result!
print("The time of crime committed during the day vs night: ")
print(day_crime.value_counts())

# create an empty dictionary for our intervals
crime_by_6hours = {'Night':'','Morning':'',"Afternoon":'', "Evening":''}
i = 0
# create a for loop that goes through the various dictionary keys applying the filter we want
# to for each 6 hour time interval
for item, val in crime_by_6hours.items():
    crime_by_6hours[item] = hour_crime[(hour_crime >= i) & (hour_crime < i+6)].shape[0]
    i += 6

# print out the results!
print("Crimes broken down by 6 hour intervals: ")
print(crime_by_6hours)

weekday_of_crime = crime_data['Dispatch Date / Time'].dt.weekday_name
weekday_of_crime_sorted = weekday_of_crime.value_counts()

print("The days of crime committed on: ")
print(weekday_of_crime_sorted)



The month's of crime committed: 
7     3520
8     4002
9     3927
10    4075
11    3941
12    3904
Name: Dispatch Date / Time, dtype: int64
The time of crime committed during the day vs night: 
True     13813
False     9556
Name: Dispatch Date / Time, dtype: int64
Crimes broken down by 6 hour intervals: 
{'Night': 3269, 'Morning': 6971, 'Afternoon': 6842, 'Evening': 6287}
The days of crime committed on: 
Tuesday      3836
Monday       3734
Wednesday    3611
Friday       3594
Thursday     3404
Saturday     2807
Sunday       2383
Name: Dispatch Date / Time, dtype: int64


When looking at the above print outs we can now make some conclusions:

The dataset only contains data from July - December. That is obviously a limitation of the data and we cannot make full assumptions. If we were to take this data seriously we would need to have many years of crime data in order to make full conclusions. 

1) However, using this data we see that rougly a similar amount of committed crimes occuring from August to December. Without more data we can only make so many conclusions, but it looks like July is either an outlier with a decrease in crime or part of the year that has less crimes being committed.

2) We have found that crimes committed during the day vs night are 13813 to 9556, it is quite obvious that the majority of crime happens during the day. A whole 31% difference! 
Now let's look at this more in depth, let's break this out into the morning, afternoon, evening, and night.

morning -> 6am - 12pm -> Total crimes committed = 6971
afternoon = 12pm - 6pm -> Total crimes committed = 6842
evening = 6pm - 12am -> Total crimes committed = 6287
night = 12am - 6am -> Total crimes committed = 3269

Now we have a better resolution of the times the crimes occur. We can see that during the night crimes are roughly cut in half, this is what we would expect as we would think most people would be sleeping between 12am - 6am, even the criminals!

3) Lastly, we want to look at what day of the week is least likely that crimes will be committed on. We can see that the least likely crimes are going to be committed on is saturday and sunday. It seems even criminals need to take a break every once in awhile...